internal/controller/standalone_pgadmin/controller.go

```diff
@@ -59,3 +116,343 @@ func (r *PGAdminReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// - https://docs.k8s.io/concepts/workloads/controllers/statefulset/#stable-network-id
+	// - https://docs.k8s.io/concepts/services-networking/dns-pod-service/#pods
+	sts.Spec.ServiceName = standalonePGAdminService(pgadminCR).Name
+
+	// Set the StatefulSet update strategy to "RollingUpdate", and the Partition size for the
+	// update strategy to 0 (note that these are the defaults for a StatefulSet).  This means
+	// every pod of the StatefulSet will be deleted and recreated when the Pod template changes.
+	// - https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#rolling-updates
+	// - https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#forced-rollback
+	sts.Spec.UpdateStrategy.Type = appsv1.RollingUpdateStatefulSetStrategyType
+	sts.Spec.UpdateStrategy.RollingUpdate = &appsv1.RollingUpdateStatefulSetStrategy{
+		Partition: initialize.Int32(0),
+	}
+
+	// Use scheduling constraints from the cluster spec.
+	sts.Spec.Template.Spec.Affinity = pgadminCR.Spec.Affinity
+	sts.Spec.Template.Spec.Tolerations = pgadminCR.Spec.Tolerations
+
+	if pgadminCR.Spec.PriorityClassName != nil {
+		sts.Spec.Template.Spec.PriorityClassName = *pgadminCR.Spec.PriorityClassName
+	}
+
+	// Restart containers any time they stop, die, are killed, etc.
+	// - https://docs.k8s.io/concepts/workloads/pods/pod-lifecycle/#restart-policy
+	sts.Spec.Template.Spec.RestartPolicy = corev1.RestartPolicyAlways
+
+	// pgAdmin does not make any Kubernetes API calls. Use the default
+	// ServiceAccount and do not mount its credentials.
+	sts.Spec.Template.Spec.AutomountServiceAccountToken = initialize.Bool(false)
+
+	// Do not add environment variables describing services in this namespace.
+	sts.Spec.Template.Spec.EnableServiceLinks = initialize.Bool(false)
+
+	// set the image pull secrets, if any exist
+	sts.Spec.Template.Spec.ImagePullSecrets = pgadminCR.Spec.ImagePullSecrets
+
+	if err := errors.WithStack(r.setControllerReference(pgadminCR, sts)); err != nil {
+		return err
+	}
+
+	Pod(pgadminCR, configmap, &sts.Spec.Template.Spec, dataVolume)
+
+	return errors.WithStack(r.apply(ctx, sts))
+}
+
+// Pod populates a PodSpec with the container and volumes needed to run pgAdmin.
+func Pod(
+	inPGAdmin *v1beta1.PGAdmin,
+	inConfigMap *corev1.ConfigMap,
+	outPod *corev1.PodSpec,
+	pgAdminVolume *corev1.PersistentVolumeClaim,
+) {
+
+	// create the pgAdmin Pod volumes
+	pgAdminData := corev1.Volume{Name: dataVolume}
+	pgAdminData.VolumeSource = corev1.VolumeSource{
+		PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+			ClaimName: pgAdminVolume.Name,
+			ReadOnly:  false,
+		},
+	}
+
+	configVolumeMount := corev1.VolumeMount{
+		Name: "pgadmin-standalone-pgadmin", MountPath: configMountPath, ReadOnly: true,
+	}
+	configVolume := corev1.Volume{Name: configVolumeMount.Name}
+	configVolume.Projected = &corev1.ProjectedVolumeSource{
+		Sources: podConfigFiles(inConfigMap, *inPGAdmin),
```

Comment:
i would think this func should exist in the `pgadmin` package - `pgadmin.PodConfigFiles`